Ginger/GingerCoreNET/ActionsLib/APIModelActionConversion/ApiActionConversionUtils.cs

```diff
@@ -0,0 +1,461 @@
+        }
+
+        /// <summary>
+        /// This method is used to check if the Parameter exists in the model, if exists then add the values to model parameter
+        /// </summary>
+        /// <param name="lstParameters"></param>
+        /// <param name="param"></param>
+        /// <returns></returns>
+        private bool CheckParameterExistsIfExistsThenAddValues(ObservableList<AppModelParameter> lstParameters, AppModelParameter param)
+        {
+            bool isExists = false;
+            foreach (var item in lstParameters)
+            {
+                if (item.ItemName.Contains(param.ItemName))
+                {
+                    isExists = true;
+                    foreach (var val in param.OptionalValuesList)
+                    {
+                        var paramOptionalValue = item.OptionalValuesList.Where(x => x.Value == val.Value).FirstOrDefault();
+                        if (!string.IsNullOrEmpty(Convert.ToString(val.Value)) && Convert.ToString(val.Value) != "?" && (paramOptionalValue == null || paramOptionalValue.Value != val.Value))
+                        {
+                            item.OptionalValuesList.Add(val);
+                        }
+                    }
+                    break;
+                }
+            }
+            return isExists;
+        }
+
+        /// <summary>
+        /// This method is used to create or update API Model from Soap action
+        /// </summary>
+        /// <param name="act"></param>
+        /// <param name="path"></param>
+        /// <returns></returns>
+        private void SetApplicationAPIModel(ref ApplicationAPIModel aPIModel, Act act, bool pullValidations)
+        {
+            string folderPath = Path.Combine(WorkSpace.Instance.Solution.ContainingFolderFullPath, @"Applications Models\API Models\");
+            try
+            {
+                aPIModel.ItemName = act.ItemName;
+                aPIModel.ContainingFolder = folderPath;
+                aPIModel.ContainingFolderFullPath = Path.Combine(folderPath, string.Format("{0}.xml", act.ItemName));
+                if (((ActWebAPIBase)act).HttpHeaders != null)
+                {
+                    if (aPIModel.HttpHeaders == null)
+                    {
+                        aPIModel.HttpHeaders = new ObservableList<APIModelKeyValue>();
+                    }
+                    foreach (var header in ((ActWebAPIBase)act).HttpHeaders)
+                    {
+                        APIModelKeyValue keyVal = new APIModelKeyValue();
+                        keyVal.ItemName = header.ItemName;
+                        keyVal.Param = header.ItemName;
+                        keyVal.FileName = header.ItemName;
+                        keyVal.Value = header.Value;
+                        aPIModel.HttpHeaders.Add(keyVal);
+                    }
+                }
+
+                if (pullValidations)
+                {
+                    aPIModel.ReturnValues = act.ReturnValues;
```

Comment:
we can't take all validations, which should take only return values with "Expected" setup or StoreTo ApplicationGlobalParam- only those are supported in API Model.
for example storeTo DS or Variable is not supported on API Model